pkg/handlers/query.go

```diff
@@ -1,22 +1,68 @@
 package handlers
 
 import (
-	"fmt"
+	"database/sql"
+	"encoding/json"
 	"net/http"
+	"reflect"
 
 	gabi "github.com/app-sre/gabi/pkg"
 )
 
+type QueryData struct {
+	Query string
+}
+
 func Query(env *gabi.Env) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		// https://stackoverflow.com/a/46021789
 		env.Logger.Info("testing")
-		fmt.Fprintf(w, "%s %s %s \n", r.Method, r.URL, r.Proto)
+		// fmt.Fprintf(w, "%s %s %s \n", r.Method, r.URL, r.Proto)
 		for k, v := range r.Header {
-			fmt.Fprintf(w, "Header field %q, Value %q\n", k, v)
+			// fmt.Fprintf(w, "Header field %q, Value %q\n", k, v)
 			env.Logger.Infof("Header field %q, Value %q", k, v)
 		}
-		fmt.Fprintf(w, "Host = %q\n", r.Host)
-		fmt.Fprintf(w, "RemoteAddr= %q\n", r.RemoteAddr)
+		// fmt.Fprintf(w, "Host = %q\n", r.Host)
+		// fmt.Fprintf(w, "RemoteAddr= %q\n", r.RemoteAddr)
+
+		var q QueryData
+		err := json.NewDecoder(r.Body).Decode(&q)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		rows, err := env.DB.Query(q.Query)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+		}
+		defer rows.Close()
+		cols, err := rows.Columns() // Remember to check err afterwards
+		vals := make([]interface{}, len(cols))
+		var result [][]string
+		var keys []string
+		for i, _ := range cols {
+			vals[i] = new(sql.RawBytes)
+			keys = append(keys, cols[i])
+		}
+		result = append(result, keys)
+		for rows.Next() {
+			err = rows.Scan(vals...)
+			// Now you can check each element of vals for nil-ness,
+			// and you can use type introspection and type assertions
+			// to fetch the column into a typed variable.
+			var row []string
+			for _, value := range vals {
+				content := reflect.ValueOf(value).Interface().(*sql.RawBytes)
+				row = append(row, string(*content))
+			}
+			result = append(result, row)
+		}
+		err = rows.Err()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+		}
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(result)
```

Comment:
This works well and easily. I think we might need to ask Jamie about return format, or perhaps even our 'customers.' I am ok with the current output format, but we might need to change it in the future to better support the customer needs. 

```
[
  [
    "actor_id",
    "first_name",
    "last_name",
    "last_update"
  ],
  [
    "1",
    "PENELOPE",
    "GUINESS",
    "2006-02-15T04:34:33Z"
  ],
   ...
```

Reply:
No changes needed for this merge, by the way. The note on output formatting review is scoped for after the MVP.